src/main/java/frc/robot/subsystems/Swerve.java

```diff
@@ -0,0 +1,275 @@
+        zeroGyro();
+        resetPosition();
+
+        odometry.resetPosition(rotation(), modulePositions(), pose);
+        odometry.resetPosition(rotation(), modulePositions(), pose);
+    }
+
+    public void resetPosition() {
+        for (KrakenSwerveModule mod : modules)
+            mod.resetDrivePosition();
+    }
+
+    public void syncEncoders() {
+        for (KrakenSwerveModule mod : modules)
+            mod.resetSteerPosition();
+    }
+
+    public void resetAbsolute() {
+        for (KrakenSwerveModule mod : modules)
+            mod.resetAbsolute();
+    }
+
+    public void resetSteerPositions() {
+        for (KrakenSwerveModule mod : modules)
+            mod.set(0, 0);
+    }
+
+    public void setModuleStates(SwerveModuleState[] states) {
+        SwerveDriveKinematics.desaturateWheelSpeeds(states, MAX_VELOCITY);
+        for (int i = 0; i < modules.length; i++) {
+            modules[i].set((states[i].speedMetersPerSecond / MAX_VELOCITY), states[i].angle.getRadians());
+        }
+    }
+
+    public ChassisSpeeds getChassisSpeeds() {
+        return chassisSpeeds;
+    }
+
+    public void enable() {
+        active = true;
+    }
+
+    public void disable() {
+        active = false;
+
+        for (KrakenSwerveModule mod : modules)
+            mod.stop();
+    }
+
+    public void periodic() {
+        SwerveModuleState[] states = kinematics.toSwerveModuleStates(chassisSpeeds);
+        if (active && chassisSpeeds != new ChassisSpeeds())
+            setModuleStates(states);
+        current_states.set(moduleStates(modules));
+        target_states.set(states);
+        Pose2d pose = odometry.update(rotation(), modulePositions());
+        posePublisher.set(pose);
+
+        SmartDashboard.putNumber("X position", pose.getX());
+        SmartDashboard.putNumber("Y position", pose.getY());
+
+        SmartDashboard.putNumber("Odometry rotation", rotation().getDegrees());
+        SmartDashboard.putNumber("Pigeon Yaw", pigeon2.getYaw().getValueAsDouble());
+        SmartDashboard.putNumber("Pigeon Pitch", pigeon2.getPitch().getValueAsDouble());
+        SmartDashboard.putNumber("Pigeon Roll", pigeon2.getRoll().getValueAsDouble());
+
+        SmartDashboard.putString("Drive Mode", (field_oritented) ? "Field-Oriented" : "Robot-Oriented");
+    }
+
+    public static final class DriveConstants {
+        public static final double DRIVETRAIN_TRACKWIDTH_METERS = Units.inchesToMeters(19.5);
+        public static final double DRIVETRAIN_WHEELBASE_METERS = Units.inchesToMeters(21.5);
+
+        public static final int[] DRIVE_ID = { 4, 2, 6, 8 }; // FL, FR, BL, BR
+        public static final int[] STEER_ID = { 5, 3, 7, 9 }; // FL, FR, BL, BR
+        public static final int[] ENCODER_ID = { 11, 10, 12, 13 }; // FL, FR, BL, BR
+        public static final String[] LAYOUT_TITLE = { "Front Left", "Front Right", "Back Left", "Back Right" };
+
+        public static final int PIGEON_ID = 14;
```

Comment:
PIGEON_ID will be 6 or 10 depending on if we can have the CANMags have the same IDs as the chassis motors.

Reply:
Just put 6 for now